fix generation of restricted dgv file

yunfeiguo opened the pull request:
- [ ] [RES-XXXX](https://binatechnologies.atlassian.net/browse/RES-XXXX)
- [ ] Blocking PRs: ADD HERE IF ANY
* People
  - [ ] Reviewers (>=1): ADD HERE
  - [ ] Merger (>=1): ADD HERE
  - [ ] FYI: ADD HERE IF ANY
- [ ] Release note (if needed)
- [ ] No code copied from outside Bina 
- [ ] Tests (Unit, Workflow, ITL & Gherkin) are passing
- [ ] PR labels, milestones & assignees

# Summary

enable generation of restricted DGV file (in the context of simulation).

marghoob commented:
@yunfeiguo did you miss pushing some commit(s)? I do not see any change apart from the check for sortedness.

yunfeiguo commented:
@marghoob other work is still in progress. 😄

yunfeiguo commented on varsim_multi.py:
```diff
@@ -83,7 +83,7 @@ def varsim_multi(reference,
             # Now generate the restricted sampled VCF for the sample
             _, [restricted_sampled_vcf] = gen_restricted_ref_and_vcfs(reference, [sampled_vcf], regions, [], os.path.join(sample_dir, "restricted_randvcf"), flank=0)
-            sample_variant_vcfs = (restricted_vcfs if index >= len(samples) else []) + [restricted_sampled_vcf]
+            sample_variant_vcfs = (restricted_vcfs if index < len(samples) else []) + [restricted_sampled_vcf]
```
Hi @marghoob, could you please check this change? Thanks.

yunfeiguo commented:
Hi @marghoob 

the real cause for #148 is actually missing `-l` option for `art`, I just fixed that by adding some options for simulator in `varsim_multi.py`. I also added capability for varsim to do restricted simulation on `dgv` file. I have tested the PR (with minimal test).

marghoob commented on src/main/java/com/bina/varsim/tools/simulation/RandVCF2VCF.java:
```diff
@@ -45,13 +47,17 @@
     int numMNP = NUM_MNP_ARG;
     @Option(name = "-num_complex", usage = "Number of complex variants (other ones) to sample [" + NUM_COMPLEX_ARG + "]")
     int numComplex = NUM_COMPLEX_ARG;
+    @Option(name = "-num_dup", usage = "Number of complex variants (other ones) to sample [" + NUM_DUP_ARG + "]")
```
So, you are reusing the ins/del params for short variants? If you are including indel SVs with the small SVs, the random sampling will most-likely end up having no indel SVs since they are very few in number.

yunfeiguo replied:
That's a good point. Here I adopted a workaround, please refer to https://github.com/bioinform/varsim/pull/149/files#diff-163a888df028d8bfa31a54c7682b7a12R112

basically, the `dgv_vcf` and `small_vcf` will be sampled independent of each other.

marghoob commented on varsim_multi.py:
```diff
@@ -207,6 +241,27 @@ def varsim_multi(reference,
+                                default=0.6,
+                                type=float)
+    dwgsim_group = main_parser.add_argument_group("DWGSIM options")
+    dwgsim_group.add_argument("--dwgsim_start_e", metavar="first_base_error_rate", help="Error rate on the first base",
```
Why did you add simulator-specific options?

yunfeiguo replied:
if we don't provide simulator-specific options, then users need to figure out what they need to use for each simulator. e.g., `art` requires `-l` for read length, but the user in #148 didn't know about this. By providing these simulator-specific options (along with reasonable default values), varsim might be more user-friendly.

please see here for previous simulator option: https://github.com/bioinform/varsim/pull/149/files#diff-163a888df028d8bfa31a54c7682b7a12R312

Thanks.

marghoob replied:
That would make it messy to support more reads simulators. I wanted to avoid having to parse simulator-specific options only to pass them to the simulator. I'd suggest not adding these simulator-specific options and instead have simulator-specific parsers which later parse `args.simulator_options` and do error-checks.

marghoob commented on varsim_multi.py:
```diff
@@ -207,6 +241,27 @@ def varsim_multi(reference,
+    art_group.add_argument("--profile_2", metavar="profile_file2", help="ART error profile for second end", default="")
+    art_group.add_argument("--art_options", help="ART command-line options", default="")
+
+    pbsim_group = main_parser.add_argument_group("PBSIM options")
```
Let us remove pbsim from the list of supported simulators. I think I had not put it in `varsim_multi.py`, so no need to add it here.

yunfeiguo replied:
Hi @marghoob I have removed simulator-specific options. Thanks.

yunfeiguo commented:
Hi @marghoob I will close this PR because all changes can be seen in #155. Besides, the changes for this PR and #155 are not readily separable.
